pylint/utils/pragma_parser.py

```
@@ -0,0 +1,145 @@
+    "(?P<{:s}>{:s})".format(token_name, token_rgx)
+    for token_name, token_rgx in TOKEN_SPECIFICATION
+)
+
+
+def emit_pragma_representer(action, messages):
+    if not messages and action in MESSAGE_KEYWORDS:
+        raise MissingMessage(
+            "The keyword is not followed by message identifier", action
+        )
+    return PragmaRepresenter(action, messages)
+
+
+class PragmaParserError(Exception):
+    """
+    A class for exceptions thrown by pragma_parser module
+    """
+
+    def __init__(self, message, token):
+        """
+        :args message: explain the reason why the exception has been thrown
+        :args token: token concerned by the exception
+        """
+        self.message = message
+        self.token = token
+        super(PragmaParserError, self).__init__(self.message)
+
+
+class UnknownKeyword(PragmaParserError):
+    """
+    Thrown in case the keyword is not recognized
+    """
+
+
+class MissingAssignment(PragmaParserError):
+    """
+    Thrown in case the = sign is missing
+    """
+
+
+class UnsupportedAssignment(PragmaParserError):
+    """
+    Throw in case the assignment sign follows a licit keyword but
+    that doesn't support assignment
+    """
+
+
+class MissingKeyword(PragmaParserError):
+    """
+    Thrown in case keyword is missing
+    """
+
+
+class MissingMessage(PragmaParserError):
+    """
+    Thrown in case message identifier is missing
+    """
+
+
+def parse_pragma(pylint_pragma: str) -> Generator[PragmaRepresenter, None, None]:
+    action = None
+    messages = []  # type: List[str]
+    assignment_required = False
+    previous_token = ""
+
+    for mo in re.finditer(TOK_REGEX, pylint_pragma):
+        kind = mo.lastgroup
+        value = mo.group()
+
+        if kind == "ASSIGN":
+            if not assignment_required:
+                if action:
+                    # A keyword has been found previously but doesn't support assignement
+                    raise UnsupportedAssignment(
```

I wonder if we need all these exceptions or if we should raise the same exception but with a corresponding message for each situation. My reasoning is that nobody will need to handle all these exceptions to verify that a pragma is broken and will result to handle just the uppermost base class. What do you think?